ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/rest/FileController.java

```diff
@@ -100,12 +100,19 @@ public void getArchiveFile(@PathVariable("id") String id,
                                @RequestParam(value = "destinationIV", required = false) String destinationIV,
                                @RequestParam(value = "startCoordinate", required = false, defaultValue = "0") long startCoordinate,
                                @RequestParam(value = "endCoordinate", required = false, defaultValue = "0") long endCoordinate,
+                               @RequestHeader(value = "Range", required = false, defaultValue = "") String range,
                                HttpServletRequest request,
                                HttpServletResponse response) {
 
         // Resolve Archive ID to actual File Path/URL - Needs Organization-Specific Implementation!
         ArchiveSource source = archiveService.getArchiveFile(id, request, response);
 
+        if (range.length() > 0 && range.startsWith("bytes=") && startCoordinate == 0 && endCoordinate == 0) {
+            String[] ranges = range.substring("bytes=".length()).split("-");
+            startCoordinate = Long.valueOf(ranges[0]);
+            endCoordinate = Long.valueOf(ranges[1]) + 1;
```

Comment:
Can you explain to me this line @anandmohan777 ? :-) (it's because I do not understand)

Reply:
Suppose if the incoming request byte range is bytes=10-300. The endCoordinate we are adding one extra byte because last bit is not excluded in FIRE
startCoordinate = 10
endCoordinate = 301

Reply:
Thanks you so much @anandmohan777 I understand now :-)